MapboxNavigation/Geometry.swift

```diff
@@ -162,6 +162,10 @@ func closestCoordinate(on polyline: [CLLocationCoordinate2D], to coordinate: CLL
         let intersectionPoint = intersection((perpendicularPoint1, perpendicularPoint2), segment)
         let intersectionDistance: CLLocationDistance? = intersectionPoint != nil ? coordinate - intersectionPoint! : nil
         
+        if distances.1 < distances.0 && roundUpToNextCoordinate {
+            index += 1
```

Comment:
For consistency with Turfjs/turf#563, can we move this increment down to the `distances.1 < closestCoordinate?.distance ?? CLLocationDistanceMax` if statement? Or should Turfjs/turf#563 gain a new if statement for consistency with this PR?